File: maui/Platforms/Android/PosSystemTransportFactory.cs
namespace fiskaltrust.Middleware.Demo.Platforms.Android
{
    public static class PosSystemTransportFactory
    {
        public static IPosSystemTransport GetInstance(bool useBoundService)
        {
            if (useBoundService)
            {
                return new BoundServiceTransport();
            }
            else {
                return new ActivityTransport();
            }

        }
    }
}

File: java/app/src/main/java/eu/fiskaltrust/middleware/demo/transport/PosSystemTransportFactory.java
package eu.fiskaltrust.middleware.demo.transport;

public final class PosSystemTransportFactory {

  private PosSystemTransportFactory() { }

  public static PosSystemTransport getInstance(boolean useBoundService) {
    return useBoundService ? new BoundServiceTransport() : new ActivityTransport();
  }
}
